## Data Cleaning Pipeline (Kaggle->Client-Ready)

# Importing Libraries 

In [44]:
import pandas as pd
import numpy as np

# Loading CSV file

In [45]:
df = pd.read_csv("../data/raw_data.csv")

In [46]:
df.shape

(20, 11)

In [47]:
df.head(5)

,Rank,Peak,All Time Peak,Actual gross,Adjusted gross (in 2022 dollars),Artist,Tour title,Year(s),Shows,Average gross,Ref.
0,1,1,2,"$780,000,000","$780,000,000",Taylor Swift,The Eras Tour †,2023–2024,56,"$13,928,571",[1]
1,2,1,7[2],"$579,800,000","$579,800,000",Beyoncé,Renaissance World Tour,2023,56,"$10,353,571",[3]
2,3,1[4],2[5],"$411,000,000","$560,622,615",Madonna,Sticky & Sweet Tour ‡[4][a],2008–2009,85,"$4,835,294",[6]
3,4,2[7],10[7],"$397,300,000","$454,751,555",Pink,Beautiful Trauma World Tour,2018–2019,156,"$2,546,795",[7]
4,5,2[4],NaN,"$345,675,146","$402,844,849",Taylor Swift,Reputation Stadium Tour,2018,53,"$6,522,173",[8]


In [48]:
df.tail(5)

,Rank,Peak,All Time Peak,Actual gross,Adjusted gross (in 2022 dollars),Artist,Tour title,Year(s),Shows,Average gross,Ref.
15,16,NaN,NaN,"$184,000,000","$227,452,347",Pink,The Truth About Love Tour,2013–2014,142,"$1,295,775",[22]
16,17,NaN,NaN,"$170,000,000","$213,568,571",Lady Gaga,Born This Way Ball,2012–2013,98,"$1,734,694",[d]
17,18,NaN,NaN,"$169,800,000","$207,046,755",Madonna,Rebel Heart Tour,2015–2016,82,"$2,070,732",[4]
18,19,NaN,NaN,"$167,700,000[e]","$204,486,106",Adele,Adele Live 2016,2016–2017,121,"$1,385,950",[25]
19,20,NaN,NaN,"$150,000,000","$185,423,109",Taylor Swift,The Red Tour,2013–2014,86,"$1,744,186",[26]


In [49]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 11 columns):
 #   Column                            Non-Null Count  Dtype
---  ------                            --------------  -----
 0   Rank                              20 non-null     int64
 1   Peak                              9 non-null      str  
 2   All Time Peak                     6 non-null      str  
 3   Actual gross                      20 non-null     str  
 4   Adjusted gross (in 2022 dollars)  20 non-null     str  
 5   Artist                            20 non-null     str  
 6   Tour title                        20 non-null     str  
 7   Year(s)                           20 non-null     str  
 8   Shows                             20 non-null     int64
 9   Average gross                     20 non-null     str  
 10  Ref.                              20 non-null     str  
dtypes: int64(2), str(9)
memory usage: 1.8 KB


In [50]:
df.describe()

,Rank,Shows
count,20.000000,20.000000
mean,10.450000,110.000000
std,5.942488,66.507617
min,1.000000,41.000000
25%,5.750000,59.000000
50%,10.500000,87.000000
75%,15.250000,134.500000
max,20.000000,325.000000


In [51]:
df.dtypes

Rank                                int64
Peak                                  str
All Time Peak                         str
Actual gross                          str
Adjusted gross (in 2022 dollars)      str
Artist                                str
Tour title                            str
Year(s)                               str
Shows                               int64
Average gross                         str
Ref.                                  str
dtype: object

In [52]:
df.columns

Index(['Rank', 'Peak', 'All Time Peak', 'Actual gross',
       'Adjusted gross (in 2022 dollars)', 'Artist', 'Tour title', 'Year(s)',
       'Shows', 'Average gross', 'Ref.'],
      dtype='str')

## DAMAGE REPORT
This dataset contains information about the top 20 highest-grossing concert tours. It includes rankings, artists, tour names, revenue, number of shows, and years.

## Dataset Shape

- Rows: **20**
- Columns: **11**

## Data Types

The dataset contains two numerical columns and nine string columns. Revenue columns are stored as text due to currency symbols and commas.

## Missing Values

Missing values are present in the **Peak** and **All Time Peak** columns. The remaining columns are complete.

## Statistical Summary

- Average Rank: **10.45**
- Average Shows: **110**
- Maximum Shows: **325**
- Minimum Shows: **41**

## Initial Observations

- Taylor Swift appears multiple times in the dataset.
- Revenue columns need cleaning before analysis.
- The number of shows varies greatly among tours.
- Some ranking information is unavailable for certain tours.